File: backend/app/schemas/skill_gap.py
from pydantic import BaseModel
from typing import List

class SkillGapRequest(BaseModel):
    missing_skills: List[str]
    role: str
    job_id: int  # for context only, not DB lookup

class LearningResource(BaseModel):
    title: str
    type: str        # "course" | "documentation" | "project idea" | "book"
    url: str
    duration: str    # e.g. "4 hours", "2 weeks"
    level: str       # "beginner" | "intermediate" | "advanced"
    why: str         # why this is relevant to the role

class SkillGapItem(BaseModel):
    skill: str
    priority: str    # "high" | "medium" | "low"
    context: str     # 1-sentence why this skill matters for the role
    resources: List[LearningResource]

class SkillGapResponse(BaseModel):
    role: str
    items: List[SkillGapItem]
    learning_path: List[str]   # ordered list: "Start with X, then Y..."

File: frontend/src/api/skillGap.ts
import axios from "axios";

const API_URL = import.meta.env.VITE_API_URL || "http://localhost:8000";

export interface LearningResource {
  title: string;
  type: "course" | "documentation" | "project idea" | "book";
  url: string;
  duration: string;
  level: "beginner" | "intermediate" | "advanced";
  why: string;
}

export interface SkillGapItem {
  skill: string;
  priority: "high" | "medium" | "low";
  context: string;
  resources: LearningResource[];
}

export interface SkillGapResponse {
  role: string;
  items: SkillGapItem[];
  learning_path: string[];
}

export const getSkillGapResources = async (
  missing_skills: string[],
  role: string,
  job_id: number
): Promise<SkillGapResponse> => {
  const token = localStorage.getItem("token");
  const response = await axios.post(
    `${API_URL}/skill-gap/resources`,
    { missing_skills, role, job_id },
    { headers: { Authorization: `Bearer ${token}` } }
  );
  return response.data;
};